api/apps/geoprocessing/src/modules/surface-cost/adapters/typeorm-available-planning-units.ts

```diff
@@ -0,0 +1,28 @@
+import { Repository } from 'typeorm';
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { ScenariosPlanningUnitGeoEntity } from '@marxan/scenarios-planning-unit';
+
+import { GetAvailablePlanningUnits } from '../ports/available-planning-units/get-available-planning-units';
+
+@Injectable()
+export class TypeormAvailablePlanningUnits
```

Comment:
hm, I get this is the point of a decoupled architecture, but realistically we're not going to use anything else than TypeORM in this context - and if we did, we would likely replace the use of TypeORM with something else in a whole adaptor, not just in part of it.

To me, this class is a plain getter of the db state of things, happening to do so via TypeORM as an implementation detail - I would avoid proliferation of naming conventions and prefixes, and rename it to something like `AvailablePlanningUnits`.

Moreover, I would invite to avoid breaking things down so finely unless there's a compelling reason to - this class' `get()` method could pretty much be `ScenariosService.getPlanningUnits()` - it could still be composed via an interface while implementing this and other interfaces in a class such as `ScenariosService`.

It may be my FP neurons, but class or interface names that use verbs (such as `GetAvailablePlanningUnits`) are a bit of a smell that we are breaking things down too aggressively (for the kind and size of domain at hand, at least).

Reply:
Prefixing `TypeORM` is one of the strategies to avoid naming implementations the same way as their interfaces.

One of the reasons behind breaking things down so finely is that to make the code more unit testable (and have a lower cognitive load due to a smaller context) – especially when we have the interface (of course, a class can implement a bunch of them). For me, during refactor is easier to join things than split.

Using verbs in names are CQRS-ish, one class - one purpose.

Reply:
@hotzevzl  how about `DbAvailablePlanningUnits`/`AvailablePlanningUnitsRepository` then? it hides implementation details (a little) but still exposes what is the aim of the adapter. It can be it could be resolved from something different.

```
 it could still be composed via an interface while implementing this and other interfaces in a class such as ScenariosService
``` 
it could be, even we got such case somewhere. The *service tho is "hard" to test and does not exist in geoprocessing context.

Verbs - as @Dyostiq stated was the intention. I agree tho to name it just `AvailablePlanningUnitsRepository`

Reply:
@kgajowy `AvailablePlanningUnitsRepository` sounds good, reuses concepts already in use and does not bind to orm/db.

And yes, I agree that almost-isolated functions are easier to mock and test, as well as an opportunity to use as precise as possible/needed typing.

I am fine with the approach - in practice, this is a way to namespace things and separate providers, but I would like to all converge on a way to do this consistently and to make this easy to read, or we end up having a bit of a patchwork of similar but slightly different rationales for breaking things down, using them and so on, which to me is a bit of a highway to the dreaded ball of mud :)

Right now, what I think I need to do even just to review a PR is:
- check the list of providers in a module
- check their interfaces (as we go on this becomes less of a cognitive load as I'd have seen most of them in earlier PRs)
- check their implementations
- check how all the pieces fit together and are used in the overall workflow of the feature

And frankly, I think this is a bit of a readability issue (with related cognitive load) mostly because we don't have a default structure for this, yet at least, and each module ends up being slightly off and ad-hoc, for problems that may not be all that unique.

I would be much more comfortable if we could align on a workflow (with all the relevant differences) equivalent to the general structure of BaseService - in its dumbness and limitations (coupling and so on), it does at least provide a very simple and understandable workflow for crud components where the base class describes all the default workflows and related lifecycle "hooks".

I am not even sure what to suggest as way forward as I am not yet seeing enough patterns emerging from recent work, from which I could propose a general default structure for feedback and alignment (happy to accept such a proposal though, as you may have something in mind already at least implicitly while you've been working on these components).

However, after having gone on a huge semi-philosophical tangent: for the issue at hand in this PR comment: let's do as suggested (`AvailablePlanningUnitsRepository`), and if we can stick to `GetSomething` for the abstract class that only implements a `get()` method, and `SomethingRepository` for the implementation, I'd say we would already have some understandable convention. I'm still not a fan of naming classes or interfaces with a verb, but if thinking about them as namespaces I think I can live with this convention :)

Reply:
@hotzevzl 
`Right now, what I think I need to do even just to review a PR is:`
not entirely. Application (without adapters implementation of `useClass`) is one main piece of given functionality and reviewing (even with test) is purely enough to know (also for newcomers) how it works. Implementation details of storage are just... details, kind of irrelevant from the product perspective. If one is interested, may jump into implementation details. It makes it a little more intuitive when getting familiar with whole feature - like reading API doc first what is available and what it does.

> check how all the pieces fit together and are used in the overall workflow of the feature

integration test is a perfect purpose to verify all the linkings and overall workflow to get familiar with "usage".

> However, after having gone on a huge semi-philosophical tangent
its always a pleasure to discuss things in such manner with skilled people :)

Reply:
fixed in https://github.com/Vizzuality/marxan-cloud/pull/225/commits/a6068c200c2bb6245a9b13cdd2fdb10f3134a821